backend/httpclient/tracing_middleware.go

```diff
@@ -0,0 +1,61 @@
+package httpclient
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/propagation"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	TracingMiddlewareName   = "tracing"
+	httpContentLengthTagKey = "http.content_length"
+)
+
+// TracingMiddleware is a middleware that creates spans for each outgoing request, tracking the url, method and response
+// code as span attributes.
+func TracingMiddleware(tracer trace.Tracer) Middleware {
```

Comment:
Was the [roundtripper](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/net/http/otelhttp/transport.go#L50) from `opentelemetry-go-contrib` considered here? Looks like this is doing the same work. `opts.Labels` is also [supported](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/net/http/otelhttp/transport.go#L50-L64)

Reply:
We didn't, but it might be something we're considering for the future. Grafana is still supporting both opentracing and opentelemetry. When opentracing support is removed it make sense to use otelhttp and the otelhttptrace packages.